### PR TITLE
DEP: bump min version of openpyxl to 3.0.0 #39603

### LINKED ISSUE #39603: DEP: Bump minimum version of openpyxl to 3.0.0
PR #39486 fixed an issue with openpyxl but only works for openpyxl >= 3.0.0. We should bump the minimum version with pandas 1.3 and remove the version-checking that was added in that PR.

### PR DESCRIPTION
**Pull request by fangchenli**
- [x] closes #39603

### CODE REVIEW

**Comment by jreback** [comment]
cc @rhshadrach

**Review by rhshadrach — CHANGES_REQUESTED** [review]
There are also some version checks in the openpyxl tests that can be removed/simplified.

File: pandas/io/excel/_openpyxl.py [hunk 1]
```diff
@@ -531,18 +530,12 @@ def _convert_cell(self, cell, convert_float: bool) -> Scalar:
         return cell.value
 
     def get_sheet_data(self, sheet, convert_float: bool) -> List[List[Scalar]]:
-        # GH 39001
-        # Reading of excel file depends on dimension data being correct but
-        # writers sometimes omit or get it wrong
```
Comment (rhshadrach) [inline pandas/io/excel/_openpyxl.py:536]: Can you move this comment down to the `if is_readonly` below, add that it only impacts read-only worksheets

**Review by rhshadrach — CHANGES_REQUESTED** [review]
Small request, rest look great!

**Review by rhshadrach — APPROVED** [review]
lgtm, thanks @fangchenli

**Comment by jreback** [comment]
thanks @fangchenli

**Pull request merged**